cf-custom-resources/test/unique-aliases-test.js

```diff
@@ -0,0 +1,122 @@
+    console.log = function () { };
+    console.error = function () { };
+  });
+  afterEach(() => {
+    console.log = origLog;
+    console.error = origError;
+  });
+
+  test("Bogus operation fails", () => {
+    const bogusType = "bogus";
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
+        return (
+          body.Status === "FAILED" &&
+          body.Reason ===
+          "Unsupported request type bogus (Log: /aws/lambda/testLambda/2021/06/28/[$LATEST]9b93a7dca7344adeb193d15c092dbbfd)"
+        );
+      })
+      .reply(200);
+    return LambdaTester(uniqueAliases.handler)
+      .event({
+        RequestType: bogusType,
+        RequestId: testRequestId,
+        ResourceProperties: {},
+        LogicalResourceId: "mockID",
+      })
+      .expectResolve(() => {
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("Delete event is a no-op", () => {
+    const requestType = "Delete";
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
+        return body.Status === "SUCCESS";
+      })
+      .reply(200);
+    return LambdaTester(uniqueAliases.handler)
+      .event({
+        RequestType: requestType,
+        RequestId: testRequestId,
+        ResourceProperties: {},
+        LogicalResourceId: "mockID",
+      })
+      .expectResolve(() => {
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  const aliasTest = (name, input, expectedOutput) => {
+    const tt = (name, reqType, input, expectedOutput) => {
+      test(name, () => {
+        const request = nock(ResponseURL)
+          .put("/", (body) => {
+            return body.Status === "SUCCESS" &&
+              body.PhysicalResourceId === "unique-aliases-mockID" &&
+              JSON.stringify(body.Data.UniqueAliases) === JSON.stringify(expectedOutput);
+          })
+          .reply(200);
+
+        return LambdaTester(uniqueAliases.handler)
+          .event({
+            RequestType: reqType,
+            RequestId: testRequestId,
+            ResourceProperties: {
+              Aliases: JSON.stringify(input), // aliases get passed as a string
+            },
+            LogicalResourceId: "mockID",
+          })
+          .expectResolve(() => {
+            expect(request.isDone()).toBe(true);
+          });
+      });
+    };
+
+    tt(`Create/${name}`, "Create", input, expectedOutput);
+    tt(`Update/${name}`, "Update", input, expectedOutput);
+  };
+
+  aliasTest("no aliases", {}, []);
+
+  aliasTest("one service", {
+    "svc1": ["svc1.com", "example.com"],
+  }, ["example.com", "svc1.com"]);
+
+  aliasTest("two services no common aliases", {
+    "svc1": ["svc1.com"],
+    "svc2": ["svc2.com"]
+  }, ["svc1.com", "svc2.com"]);
+
+  aliasTest("two services with a common alias", {
+    "svc1": ["svc1.com", "example.com"],
+    "svc2": ["svc2.com", "example.com"]
+  }, ["example.com", "svc1.com", "svc2.com"]);
+
+  aliasTest("two services with a common alias", {
```

Comment:
This test is a duplicate of the one above it

Reply:
whoops. i just really wanted to make sure it works lol. good catch.